contracts/middleware/BaseRemove.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.24;
+
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {BalanceDelta, BalanceDeltaLibrary} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {Proxy} from "@openzeppelin/contracts/proxy/Proxy.sol";
+import {BaseMiddleware} from "./BaseMiddleware.sol";
+import {BaseHook} from "../BaseHook.sol";
+import {BalanceDeltaLibrary} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
+import {Hooks} from "@uniswap/v4-core/src/libraries/Hooks.sol";
+import {TransientStateLibrary} from "@uniswap/v4-core/src/libraries/TransientStateLibrary.sol";
+
+abstract contract BaseRemove is BaseMiddleware {
+    using TransientStateLibrary for IPoolManager;
+
+    error HookPermissionForbidden(address hooks);
+    error HookModifiedDeltas();
+    error FailedImplementationCall();
+
+    bytes internal constant ZERO_BYTES = bytes("");
+    uint256 public constant GAS_LIMIT = 5_000_000;
+    uint256 public constant MAX_BIPS = 10_000;
+
+    // use this hookdata to override checks to save gas. keccak256("override") - 1
+    bytes32 constant OVERRIDE_BYTES = 0x23b70c8dec38c3dec67a5596870027b04c4058cb3ac57b4e589bf628ac6669e7;
+
+    uint256 public immutable maxFeeBips;
```

Comment:
why declare this variable in this contract? seems like its never used here, and actually the implementing contract would set this value?

Reply:
i was thinking it was better for standardization. if a third party calls a middleware and wants to read its maxFee, it would return zero. maybe i could make it constant?